common/src/main/java/org/keycloak/common/util/PemUtils.java

```diff
@@ -17,31 +17,28 @@
 
 package org.keycloak.common.util;
 
-import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
-
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.StringWriter;
 import java.security.Key;
-import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.security.PrivateKey;
 import java.security.PublicKey;
 import java.security.cert.Certificate;
 import java.security.cert.X509Certificate;
 
+import org.keycloak.common.crypto.CryptoIntegration;
+
 /**
  * Utility classes to extract PublicKey, PrivateKey, and X509Certificate from openssl generated PEM files
  *
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
-public final class PemUtils {
+public class PemUtils {
 
     public static final String BEGIN_CERT = "-----BEGIN CERTIFICATE-----";
     public static final String END_CERT = "-----END CERTIFICATE-----";
 
-    private PemUtils() {
+    static {
+        CryptoIntegration.init(ClassLoader.getSystemClassLoader());
```

Comment:
Same comment as for static initilization in `CertificateUtils`

Reply:
Added @Deprecated to static classes, I would prefer to open follow-up issue to remove static calls.